src/HttpClient/middlewares/cache.ts

```diff
@@ -199,24 +212,33 @@ export const cacheMiddleware = ({ type, storage }: CacheOptions) => {
         : data
 
       const expiration = Date.now() + (maxAge - currentAge) * 1000
-      await storage.set(setKey, {
-        etag,
-        expiration,
-        response: {data: cacheableData, headers, status},
-        responseEncoding,
-        responseType,
-      })
 
-      span?.log({
-        event: HttpLogEvents.LOCAL_CACHE_SAVED,
-        [HttpCacheLogFields.CACHE_TYPE]: cacheType,
-        [HttpCacheLogFields.KEY_SET]: setKey,
-        [HttpCacheLogFields.AGE]: currentAge,
-        [HttpCacheLogFields.ETAG]: etag,
-        [HttpCacheLogFields.EXPIRATION_TIME]: (expiration - Date.now())/1000,
-        [HttpCacheLogFields.RESPONSE_ENCONDING]: responseEncoding,
-        [HttpCacheLogFields.RESPONSE_TYPE]: responseType,
-      })
+      const cacheWriteSpan = createCacheSpan(cacheType, 'write', tracer, span)
+      try {
+        await storage.set(setKey, {
+          etag,
+          expiration,
+          response: {data: cacheableData, headers, status},
+          responseEncoding,
+          responseType,
+        })
+
+        span?.log({
+          event: HttpLogEvents.LOCAL_CACHE_SAVED,
+          [HttpCacheLogFields.CACHE_TYPE]: cacheType,
+          [HttpCacheLogFields.KEY_SET]: setKey,
+          [HttpCacheLogFields.AGE]: currentAge,
+          [HttpCacheLogFields.ETAG]: etag,
+          [HttpCacheLogFields.EXPIRATION_TIME]: (expiration - Date.now())/1000,
+          [HttpCacheLogFields.RESPONSE_ENCONDING]: responseEncoding,
+          [HttpCacheLogFields.RESPONSE_TYPE]: responseType,
+        })
+      } catch (error) {
+        ErrorReport.create({ originalError: error }).injectOnSpan(cacheReadSpan)
```

Comment:
Since we're talking about writing here:
```suggestion
        ErrorReport.create({ originalError: error }).injectOnSpan(cacheWriteSpan)
```

Reply:
Ooooh good catch